docs/adr/adr-0026-introducing-prettier.md

```diff
@@ -0,0 +1,176 @@
+
+## Formatters study
+
+Before choosing a formatter we had to test some in order to find the one that would suit us best.
+You'll find below the ones that we tried and some explanations....
+
+### Biome
+
+Biome is a toolchain written in Rust that can both format and lint but also more. See [Biome's website](https://biomejs.dev/) for more information.
+
+#### Pros
+
+- Biome is multipurpose, it has a formatter, a linter and more.
+
+#### Cons
+
+- It is opiniated, which means you can't really override things to match your need.
+- It only has some default agnostic formatting options.
+- While being multipurpose is a pro, it's also a con. If you want to use only one feature, you'll still need to get the whole package and deactivate the features you don't need.
+
+While we could have chosen to go for Biome as a formatter, and maybe go even further and choose it as a linter too, there's a few reasons while we decided not to go for this solution.
+- As a linter, there's not a lot of options and it can't check our CSS, so that's a clear no go.
+- If we wanted to only go for the formatter, we still had to get everything and deactivate features we didn't need/want. (formatter, analyzer..)
+- Biome has the same philosophy as Prettier for the formatting but Prettier is more known and used for now.
+
+### dprint
+
+From [dprint's website](https://dprint.dev/):
+
+> A pluggable and configurable code formatting platform written in Rust.
+
+#### Pros
+
+- Lot of options.
+- You can develop plugins.
+- If you want to develop plugins, there's a rust SDK.
+
+#### Cons
+
+- Even though it's possible to write plugins in another language, resources about this are sparse. It means you better be proficient with Rust.
+
+### Prettier
+
+[Prettier](https://prettier.io) is an opiniated code formatter.
+
+#### Pros
+
+- Can format both JS, CSS, template literals in one take.
+
+#### Cons
+
+- There's not a lot of options.
+  - It is justified by the `opiniated formatter` logic.
+
+Prettier is the option we chose to use, you can find more information about the why on the dedicated section below.
+
+### js-beautify
+
+#### Pros
+
+- Has integrations with IDEs
+
+#### Cons
+
+- Unsure about the future of the maintenance ([See github README](https://github.com/beautifier/js-beautify?tab=readme-ov-file#contributors-needed))
+- Can over-format
+
+### Others
+
+- While we knew all these tools existed we wanted to know what Lit users were using
+  - Knowing that we asked the Lit discord's what they were using
+    - Someone told us that they were using the new `Stylistic` package for ESLint that keeps formatting rules
+    - They were also using a Prettier plugin for CSS
+    - While it seemed a good idea at first, you'll find why it wasn't adapted to our case below.
+
+## Investigation after Lit Discord's question (TODO: remove this)
+
+- I have asked the Lit Discord's to know if some users used specific tools to format their Lit components
```

Comment:
**nitpick:** I would stick with "We" since it's what we use everywhere and I don't think we ever go for "I" within ADRs but it's not very important so you decide of course :wink: 
```suggestion
- We have asked the Lit Discord to know if some users used specific tools to format their Lit components
```
or
```suggestion
- We have asked the Lit's Discord to know if some users used specific tools to format their Lit components
```